### PR TITLE
remove ionizer `None`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
followup of #1237

ionizer `None` is useless since we have a clean way to call only
algorithm which are supported by an specie

- [x] rebase against #1237

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
ready for review

File: src/picongpu/include/simulationControl/MySimulation.hpp [hunk 1]
```diff
@@ -63,6 +63,7 @@
 #include "particles/InitFunctors.hpp"
 #include "particles/memory/buffers/MallocMCBuffer.hpp"
 #include "particles/traits/FilterByFlag.hpp"
+#include "particles/traits/FilterByFlag.hpp"
```
Comment (erikzenker) [inline src/picongpu/include/simulationControl/MySimulation.hpp]: Why this double include ?

Comment (psychocoderHPC) [inline src/picongpu/include/simulationControl/MySimulation.hpp]: Because it's wrong^^, thx for spotting

**Comment by psychocoderHPC** [comment]
please review

**Pull request merged**